src/foundation.cs

```diff
@@ -8752,7 +8753,7 @@ partial interface NSItemProvider : NSCopying {
 
 		[Mac (10,12)][Async]
 		[Export ("registerCloudKitShareWithPreparationHandler:")]
-		void RegisterCloudKitShare ([BlockCallback] Action<CloudKitRegistrationPreparationHandler> preparationHandler);
+		void RegisterCloudKitShare (CloudKitRegistrationPreparationAction preparationHandler);
```

Comment:
So here's the warning I was trying to fix:

warning BI1116: bmac: The parameter 'obj' in the delegate 'System.Action`1[[Foundation.CloudKitRegistrationPreparationHandler, temp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]]' does not have a [CCallback] or [BlockCallback] attribute. Defaulting to [CCallback]. Declare a custom delegate instead of using System.Action / System.Func and add the attribute on the corresponding parameter.